app/src/main/java/fr/neamar/kiss/forwarder/OreoShortcuts.java

```diff
@@ -14,14 +10,12 @@
 
     void onCreate() {
         // Shortcuts in Android O
-        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-            Intent intent = mainActivity.getIntent();
-            if (intent != null) {
-                final String action = intent.getAction();
-                if (LauncherApps.ACTION_CONFIRM_PIN_SHORTCUT.equals(action)) {
-                    new SaveOreoShortcutAsync(mainActivity, intent).execute();
-                }
-            }
+        if (prefs.getBoolean("first-run-shortcuts", true) &&
+                ShortcutUtil.areShortcutsEnabled(mainActivity)) {
+
+            ShortcutUtil.buildShortcuts(mainActivity);
```

Comment:
I can't help but think that the name is slightly ambiguous, as this will remove all shortcuts and recreate them. Maybe `rebuildShortcuts()`? Or `reinitializeShortcuts()`?

Reply:
Yep completly agree.